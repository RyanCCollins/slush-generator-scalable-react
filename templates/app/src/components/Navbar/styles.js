import styled from 'styled-components';
import Menu from 'grommet-udacity/components/Menu';

export const StyledMenu = styled(Menu)`
  flex-direction: row !important;
  flex-grow: 1 !important;
  align-items: center;
  justify-content: flex-end;
`;

export const StyledLogo = styled.img`
  max-height: 45px;
  margin-left: 6%;
`;

export const LogoPlaceholder = styled.div`
  width: 177px;
  height: 45px;
  background-color: transparent;
`;
