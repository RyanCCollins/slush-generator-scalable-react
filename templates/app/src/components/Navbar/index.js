import React from 'react';
import Header from 'grommet-udacity/components/Header';
import Title from 'grommet-udacity/components/Title';
import Menu from 'grommet-udacity/components/Menu';
import Anchor from 'grommet-udacity/components/Anchor';
import Search from 'grommet-udacity/components/Search';
import LogoImage from './logo.png';

import styles from './index.module.scss';
import cssModules from 'react-css-modules';

const Navbar = () => (
  <div className={styles.navbar}>
    <Header justify="between">
      <Title>
        <img className={styles.logo} src={LogoImage} alt="logo"/>
      </Title>
      <Menu direction="row" align="center" responsive={false}>
        <Anchor href="#" className="active">
          First
        </Anchor>
        <Anchor href="#">
          Second
        </Anchor>
        <Anchor href="#">
          Third
        </Anchor>
        <Search dropAlign={{ right: 'right' }} />
      </Menu>
    </Header>
  </div>
);

export default cssModules(Navbar, styles);
