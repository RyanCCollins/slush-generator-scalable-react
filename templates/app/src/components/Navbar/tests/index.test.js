import { shallow } from 'enzyme';
import React from 'react';
import { shallowToJson } from 'enzyme-to-json';
import Navbar from '../index';

describe('<Navbar />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Navbar />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
