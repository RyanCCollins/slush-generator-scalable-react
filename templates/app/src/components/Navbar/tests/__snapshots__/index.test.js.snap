exports[`<Navbar /> should render with default props 1`] = `
<div>
  <Header
    align="center"
    direction="row"
    justify="between"
    pad={
      Object {
        "between": "small",
        "horizontal": "none",
        "vertical": "none"
      }
    }
    responsive={false}>
    <Title
      responsive={true}>
      <img
        alt="logo"
        src={Object {}} />
    </Title>
    <Menu
      align="center"
      closeOnClick={true}
      direction="row"
      dropAlign={
        Object {
          "left": "left",
          "top": "top"
        }
      }
      pad="none"
      responsive={false}>
      <Anchor
        className="active"
        href="#"
        tag="a">
        First
      </Anchor>
      <Anchor
        href="#"
        tag="a">
        Second
      </Anchor>
      <Anchor
        href="#"
        tag="a">
        Third
      </Anchor>
      <Search
        align="left"
        dropAlign={
          Object {
            "right": "right"
          }
        }
        iconAlign="end"
        inline={false}
        responsive={true} />
    </Menu>
  </Header>
</div>
`;
